subprojects/core/src/integTest/groovy/org/gradle/api/UndefinedBuildExecutionIntegrationTest.groovy

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api
+
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.api.plugins.HelpTasksPlugin
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.file.TestFile
+import spock.lang.Unroll
+
+import java.nio.file.Files
+
+class UndefinedBuildExecutionIntegrationTest extends AbstractIntegrationSpec {
+    def undefinedBuildDirectory = Files.createTempDirectory("gradle").toFile()
+
+    def setup() {
+        executer.inDirectory(undefinedBuildDirectory)
+        executer.ignoreMissingSettingsFile()
+    }
+
+    @Unroll
+    def "shows deprecation warning when executing #task task in undefined build"() {
+        expect:
+        executer.expectDeprecationWarning("Executing Gradle tasks as part of an undefined build has been deprecated. This will fail with an error in Gradle 7.0.")
+        succeeds(task) // should fail for all tasks in Gradle 7.0
+        // In Gradle 7.0, we should test the .gradle folder is not created
+
+        where:
+        task << [ProjectInternal.HELP_TASK, ProjectInternal.PROJECTS_TASK, ProjectInternal.TASKS_TASK, HelpTasksPlugin.COMPONENTS_TASK, HelpTasksPlugin.DEPENDENCIES_TASK, HelpTasksPlugin.DEPENDENT_COMPONENTS_TASK, HelpTasksPlugin.MODEL_TASK, HelpTasksPlugin.OUTGOING_VARIANTS_TASK, HelpTasksPlugin.PROPERTIES_TASK]
```

Comment:
I think just running a single task here is OK.  There's nothing special about any of these tasks.